### ISSUE TITLE
[Bug] T.tile binary ops: multi-dim BufferLoad S[1,3] silently uses flat index 1 (S[0,1])

### ISSUE THREAD

**Issue by fredrekelthen** [tile-ai/tilelang-ascend#1679]
## Bug Description

`T.tile.max` (and every other binary op sharing `binary_op`) silently ignores all but the first index when a **multi-dimensional** buffer element (`s_ub[1, 3]`) is passed as the src1 scalar. The op behaves as if the value were `s_ub.flatten()[1]` (i.e. `s_ub[0, 1]`) instead of `s_ub[1, 3]`.

Root cause: `binary_op` in `tilelang/language/ascend_tile.py` forwards only `indices_1[0]` to the codegen (`GetValue(indices[0])`), so any index beyond the first is dropped without an error.

This is a user-facing footgun: `dst[i] = max(src0[i], S[1,3])` silently computes `max(src0[i], S.flatten()[1])`.

## Environment

**Hardware:**
- NPU Device: Ascend 910B3

**Software:**
- OS: Ubuntu (container)
- CANN version: 8.5.2
- torch-npu version: torch_npu (with torch 2.x)
- Python version: 3.11
- TileLang-Ascend version: baseline `bcd9aeb5` + doc/test-only branches (no front-end changes)
- Installation method: build from source

## Code to Reproduce

```python
import torch
import tilelang
import tilelang.language as T

PASS_CONFIGS = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}


@T.prim_func
def kernel(A: T.Tensor((4, 64), "float16"), S: T.Tensor((2, 8), "float16"), C: T.Tensor((4, 64), "float16")):
    with T.Kernel(1, is_npu=True) as (cid, _):
        a_ub = T.alloc_ub((4, 64), "float16")
        s_ub = T.alloc_ub((2, 8), "float16")
        c_ub = T.alloc_ub((4, 64), "float16")
        T.copy(A, a_ub)
        T.copy(S, s_ub)
        T.tile.max(c_ub, a_ub, s_ub[1, 3])  # intended scalar: S[1,3]
        T.copy(c_ub, C)


if __name__ == "__main__":
    torch.manual_seed(0)
    a = torch.randn(4, 64, dtype=torch.float16)
    s = torch.randn(2, 8, dtype=torch.float16)

    func = tilelang.compile(kernel, out_idx=[-1], pass_configs=PASS_CONFIGS, target="ascendc")
    out = func(a.npu(), s.npu())
    torch.npu.synchronize()

    ref_first_index = torch.max(a, s.view(-1)[1])  # what the code actually used
    ref_true_index = torch.max(a, s[1, 3])         # what was requested
    print("out == max(a, S.flatten()[1]):", torch.allclose(out.cpu(), ref_first_index))
    print("out == max(a, S[1,3]):         ", torch.allclose(out.cpu(), ref_true_index))
```

## Error Message

No error is raised. Both backends silently produce the wrong result:

```
[ascendc] out == max(a, S.flatten()[1]) (first index only): True
[ascendc] out == max(a, S[1,3]) (asked index):               False
[ascendc] S.flatten()[1] = 1.7646, S[1,3] = 0.6372
[pto]     out == max(a, S.flatten()[1]) (first index only): True
[pto]     out == max(a, S[1,3]) (asked index):               False
```

## Expected Behavior

- Either correctly resolve the multi-dimensional element access (`s_ub[1, 3]` -> flat index `1*8+3`), or
- Raise a clear error at trace time stating that only 1D single-index BufferLoad is supported (mirroring the documented constraint for `T.tile.max/min`).

## Actual Behavior

`S[1, 3]` silently uses the flattened index `1` (element `S[0, 1]`); results are wrong without any warning. Same behavior on both `ascendc` and `pto` backends, for every binary op sharing `binary_op` (add/sub/mul/div/max/min/bitwise_* etc.).

## Additional Context

**Programming Mode:**
- [x] Developer mode (automatic sync/buffer management)
- [ ] Expert mode (manual T.Scope/T.barrier_all)
- [ ] Mixed mode

**Backend:**
- [x] Ascend C & PTO backend
- [ ] AscendNPU IR backend

**Pass Configs (if applicable):**
```python
pass_configs = {
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_AUTO_SYNC: True,
    tilelang.PassConfigKey.TL_ASCEND_MEMORY_PLANNING: True,
}
```

**Operator Type:**
Element-wise vector binary ops (`T.tile.max/min`; shared front-end `binary_op` affects `T.tile.add/sub/mul/div/max/min` and bitwise ops).

**Additional Information:**
- Have you checked the [Programming Guide](../docs/TileLang-Ascend%20Programming%20Guide.md)? Yes
- Have you searched existing issues? Yes — no existing issue covers multi-dim BufferLoad index dropping. #1265 (slice producing BufferLoad) and #1351 (bounded-slice parse error) are related but different.
- Any other context or screenshots? `binary_op` at `tilelang/language/ascend_tile.py` (see `indices_1[0]`).
